src/engraving/playback/playbackcontext.cpp

```diff
@@ -293,6 +303,51 @@ void PlaybackContext::handleAnnotations(const ID partId, const Segment* segment,
     }
 }
 
+void PlaybackContext::handleMeasureRepeats(const std::vector<const MeasureRepeat*>& measureRepeats, const int tickPositionOffset)
+{
+    for (const MeasureRepeat* mr : measureRepeats) {
+        const Measure* currMeasure = mr->firstMeasureOfGroup();
+        if (!currMeasure) {
+            continue;
+        }
+
+        int measureRepeatTick = mr->tick().ticks() + tickPositionOffset;
+
+        for (int num = 0; num < mr->numMeasures(); ++num, currMeasure = currMeasure->nextMeasure()) {
+            const Measure* referringMeasure = mr->referringMeasure(currMeasure);
+            if (!referringMeasure) {
+                continue;
+            }
+
+            int startTick = referringMeasure->tick().ticks();
+            int endTick = referringMeasure->endTick().ticks();
+
+            copyDynamicsInRange(startTick, endTick, measureRepeatTick);
+        }
+    }
+}
+
+void PlaybackContext::copyDynamicsInRange(const int rangeStartTick, const int rangeEndTick, const int copyStartTick)
+{
+    auto startIt = m_dynamicsMap.find(rangeStartTick);
+    if (startIt == m_dynamicsMap.end()) {
+        return;
+    }
+
+    auto endIt = mu::findLess(m_dynamicsMap, rangeEndTick);
+    if (endIt != m_dynamicsMap.end()) {
+        endIt = std::next(endIt);
+    }
+
+    DynamicMap newDynamics;
+    for (auto it = startIt; it != endIt; ++it) {
+        int tick = copyStartTick + it->first;
```

Comment:
```suggestion
        int tick = copyStartTick + (it->first - rangeStartTick);
```
to make it work correctly if the original measure is not at the start of the score

Reply:
Thanks! Fixed